gateway/services/pty_manager.py

```
@@ -0,0 +1,376 @@
+            ):
+                raise RuntimeError("Failed to create process streams")
+
+            self._stdin = self._process.stdin
+            self._stdout = self._process.stdout
+            self._stderr = self._process.stderr
+            self._attached = True
+
+            logger.info("Successfully attached to session: %s", self._session_id)
+
+        except Exception as e:
+            logger.error("Failed to attach to session %s: %s", self._session_id, e)
```

Ruff (0.15.0)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)